tests/test_gitlab_pipeline.py

```diff
@@ -0,0 +1,310 @@
+from hamcrest import assert_that, empty, not_, none, equal_to, is_not, \
+    all_of, has_item
+from loguru import logger
+
+GITLAB_CONFIG_ERROR_MESSAGE = """
+
+##### GITLAB credentials not setup/configured #####
+
+# When running from Github Actions
+
+You need to setup a `GITLAB_TOKEN` secret key.
+
+# When running from your development machine (assuming you have access to the project's GitLab testing repos)
+
+Please refer to https://python-gitlab.readthedocs.io/en/stable/cli.html#content
+for more info.
+
+Here is the approach we used:
+
+- Credentials must be written in the `secrets/gitlab-python.cfg` file (which has
+been added to the `.gitignore` file so it is excluded from the repo).
+
+- The server used is the one declared as `default`.
+
+- Here is an example of `secrets/gitlab-python.cfg` file:
+```
+[global]
+default = gitlab_com
+ssl_verify = true
+timeout = 5
+
+[gitlab_com]
+url = https://gitlab.com
+private_token = YOUR_ACCESS_TOKEN
+```
+
+"""
+
+GIT_CONFIG_ERROR_MESSAGE = """
+
+##### GIT credentials not setup/configured #####
+
+# When running from Github Actions
+
+You need to setup a `GITLAB_TOKEN` secret key.
+
+# When running from your development machine (assuming you have access to the project's GitLab testing repos)
+
+Please refer to https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage
+for more info.
+
+Here is the approach we used:
+
+- use the 'store' mode by running the command line:
+```
+$ git config --global credential.helper 'store --file ~/.git-credentials'
+```
+
+- update your `~/.git-credentials` with: `https://YOUR_USERNAME:YOUR_ACCESS_TOKEN@gitlab.com`
+
+"""
+
+
+def test_gitlab_pipeline(tmp_path):
+    project_long_title = "Python library project generator GitLab pipeline test"
+    project_slug = project_long_title.lower().replace(" ", "_").replace("-", "_")
+    project_name = project_slug.replace("_", "-")
+
+    gl_client = initialise_gitlab_client()
+    gl_project = gl_client.projects.get(
+        f"opinionated-digital-center/testing-area/{project_name}"
+    )
+    repo = initialise_git_repo(gl_project.http_url_to_repo, tmp_path / project_slug)
+
+    start_time = datetime.now()
+    start_time_str = start_time.strftime("%y-%m-%d_%Hh%M-%S_%f")
+
+    branch = pull_and_prepare_branch(repo, start_time_str)
+
+    cookiecutter(
+        ".",
+        extra_context={
+            "project_name": "ODC Python boilerplate GitLab",
+            "project_slug": project_slug
+        },
+        no_input=True,
+        default_config=True,
+        overwrite_if_exists=True,
+        output_dir=tmp_path,
+    )
+
+    commit_changes_and_push(repo, branch, start_time_str)
+
+    merge_request = gl_project.mergerequests.create(
+        {
+            "source_branch": branch.name,
+            "target_branch": "master",
+            "title": f"Test on {start_time.strftime('%Y-%m-%d %H:%M:%S.%f')}",
+        }
+    )
+
+    mr_pipeline_id = verify_merge_request_pipeline_created(merge_request)
```

Comment:
https://martinfowler.com/bliki/CommandQuerySeparation.html
Dans l'idéal il faudrait plutôt avoir une fonction qui récupère ce qu'il faut, et une autre qui fait l'assertion. Mais tu peux laisser comme ça, c'est pas très grave pour le moment, c'est surtout pour partager ça en général avec toi que j'écris ce commentaire.

Reply:
Yes, especially since the verification *is* the fact that a pipeline is found... But we can discuss!